### ISSUE TITLE
Lost desktop icons

### ISSUE THREAD

**Issue by sickneet** [amirzaidi/Launcher3#588]
I often lost my icons when I update this launcher app or use system function to back up my data.
I have hundreds of apps so it's difficult for me to drag apps' icons to my screen and rearrange my icons order. What's worse, I can't remember all my icons arrangement, which made me heartache. When I reorder my icons, I realized that something is actually different from the old one.
I hope that there can be a backup function for desktop icons, and I can import the backup to the launcher app to recover all my icons!

**Comment by amirzaidi** [comment]
> I often lost my icons when I update this launcher app

This is not supposed to happen, can you give more details?

**Comment by sickneet** [comment]
> > I often lost my icons when I update this launcher app
> 
> This is not supposed to happen, can you give more details?

1.  When I backed up my data, desktop settings and the order of icons in MIUI system, it went wrong. Back up one app, disappear one desktop icon.

2.  And recently, I wanted to update this launcher app, downloading apk and installing it. After installation, I had two launchers and I could switch to the old one and the new launcher well. It went on well until the new launcher notified me to install the Pixel Bridge. I followed it and installed the bridge. Then, I found the old desktop launcher was disappeared and the new one had no icons on my screen. I had to spend hours to drag apps' icons to screen and reorder them, which made me mental breakdown.

I hope this app can back up its icons' data and I can use this data to recover all my icons and icons' order. Thank you!

**Comment by amirzaidi** [comment]
> And recently, I wanted to update this launcher app, downloading apk and installing it. After installation, I had two launchers and I could switch to the old one and the new launcher well. It went on well until the new launcher notified me to install the Pixel Bridge. I followed it and installed the bridge. Then, I found the old desktop launcher was disappeared and the new one had no icons on my screen. I had to spend hours to drag apps' icons to screen and reorder them, which made me mental breakdown.

This was because of the package name change, and will never happen again in the future (unless I make a terrible mistake somewhere). Every update from 3.8 forward should keep your current layout without having to redo it.

> When I backed up my data, desktop settings and the order of icons in MIUI system

This is a known bug with MIUI's system backup and I recommend not using it. Instead, Google's app data restore should automatically set up your homescreen if you downloaded the app from the Play Store. Not sure if that is actually working, I heard conflicting reports about it.

**Comment by Testato** [comment]
i'm in the same situation.
I'm on the 3.7 version and want update to the lòast 3.9 version.
How i can backup/restore my icons ?

**Comment by rolebu** [comment]
I often have icons disappear from a folder. For a while they also disappeared from the app list. I changed the settings and they no longer disappear form the app list. But, still disappear from the folder. If one icon is left the folder disappears, too and you have create a new and name it. 
I use to have a Pixel and really like the Launcher. I now use a LG G7 ThinQ v8.0 only because I got a great deal on trade when my wife's Pixel developed speaker problems. 
When the deal runs out, I'll be looking at Pixel again.

**Comment by amirzaidi** [comment]
Are they installed on a SD card @rolebu?

**Comment by rolebu** [comment]
No, everything is installed in Internal phone memory. The LG G7 ThinQ will not allow a move there, if I wanted to move the app or it's icons to the SD card.

Doesn't makes sense to move a Lancher there. I have a few Apps on SD, but not Lauchers.


--- LG G7 ThinQ v8.0 / Outlook v2.2

________________________________

**Comment by amirzaidi** [comment]
Try moving all of your apps to the internal storage, Google's Launcher3 sources don't account for SD cards and I haven't made any fixes for it because I don't have a phone with SD card support to test it on.
